src/Models/UserIdentityModel.php

```diff
@@ -59,6 +59,8 @@ public function create($data): void
      */
     public function createEmailIdentity(User $user, array $credentials): void
     {
+        assert($user->id !== null, '"$user->id" must not be null.');
```

Comment:
Any reason not to use an Assert library to make these production-level assertions?

Reply:
What is an Assert library?

Reply:
It's basically the same as `assert()` but with same nice built-ins and (mostly importantly) it will throw exceptions in production. They also play nice with static analysis. These are the two ones I know about: Webmozart is so popular I would guess that it is probably already in most Composer vendor directories due to dependencies. 
* https://packagist.org/packages/webmozart/assert
* https://github.com/beberlei/assert

Reply:
Thank you. I have seen webmozart/assert.

> Any reason not to use an Assert library to make these production-level assertions?

- If the only goal is to have the error occur even in a production environment, this can be accomplished with the PHP configuration. Also, if an error should be raised in production environment, an exception should be thrown, not `assert()`.
- One should be cautious about increasing the number of dependent packages in a CI project. We cannot easily add more dependent libraries.
- The current code base does not make heavy use of assertions that an assertion library is necessary, and there are currently no plans to do so.

At the moment, I do not see the need to introduce an Assertion library.